src/shared/common/Button.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import * as React from 'react';
+
+import { Component } from 'react';
+import { Text, View, StyleSheet, Pressable } from 'react-native';
+
+// loosely adapted from https://github.com/APSL/react-native-button
+const styles = StyleSheet.create({
+    button: {
+        height: 44,
+        flexDirection: 'row',
+        alignItems: 'center',
+        borderWidth: 1,
+        borderRadius: 8,
+        marginBottom: 10,
+        alignSelf: 'stretch',
+        justifyContent: 'center',
+    },
+    text: {
+        fontSize: 16,
+        lineHeight: 21,
+        fontWeight: 'bold',
+        letterSpacing: 0.25,
+        color: 'white',
+    },
+    opacity: {
+        opacity: 0.5,
+    },
+});
+
+type Props = {
+    children: any,
+    onPress: () => any,
+    textStyle: Object,
+    style: Object,
```

Comment:
There are stricter types for this, like in https://github.com/mapswipe/mapswipe/blob/8d0f2f0ab64bf45dbbf7d69b8a74807107a90c68/src/shared/views/ChangeDetection/Tile.js#L13
or
https://github.com/mapswipe/mapswipe/blob/8d0f2f0ab64bf45dbbf7d69b8a74807107a90c68/src/shared/common/SatImage.js#L3

Reply:
good catch, changed it to ViewStyleProp for Button and TextStyleProp for inner Button text